Asset Uploader UI Component/Code/Asset Uploader/Objective-C/AssetUploader/ViewController.m

```diff
@@ -0,0 +1,298 @@
+{
+    [self dismissViewControllerAnimated:YES completion:nil];
+    NSLog(@"Asset Uploader failed with error: %@", error);
+    NSString *message = [NSString stringWithFormat:@"Error: %@", error];
+    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"Upload Error"
+                                                                             message:message
+                                                                      preferredStyle:UIAlertControllerStyleAlert];
+    
+    UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil];
+    [alertController addAction:okAction];
+    [self presentViewController:alertController animated:YES completion:nil];
+}
+
+- (void)assetUploaderViewControllerDidClose:(AdobeUXAssetUploaderViewController *)assetUploader
+{
+    [self dismissViewControllerAnimated:YES completion:nil];
+    NSLog(@"Asset Uploader was dismissed without selectiong a destination folder.");
+}
+
+#pragma mark - AdobeLibraryDelegate
+
+- (void)syncFinished
+{
+    // AdobeLibraryManager completed sync, hence deregister as delegate so that AdobeLibraryManager shutsdown.
+    [[AdobeLibraryManager sharedInstance] deregisterDelegate:self];
+}
+
+#pragma mark - Priavte methods
+
+- (void)setupAdobeLibraryManager:(AdobeLibraryDownloadPolicyType)downloadPolicy
+{
+    // Below is the setup for configure & start AdobeLibraryManager.
+    // For more info regarding libraries please refer: https://creativesdk.adobe.com/docs/ios/#/articles/libraries/index.html.
+    AdobeLibraryDelegateStartupOptions *startupOptions = [[AdobeLibraryDelegateStartupOptions alloc] init];
+
+    startupOptions.autoDownloadPolicy = downloadPolicy;
+    startupOptions.autoDownloadContentTypes = @[kAdobeMimeTypeJPEG,
+                                                kAdobeMimeTypePNG];
+    startupOptions.elementTypesFilter = @[AdobeDesignLibraryColorElementType,
+                                          AdobeDesignLibraryColorThemeElementType,
+                                          AdobeDesignLibraryCharacterStyleElementType,
+                                          AdobeDesignLibraryBrushElementType,
+                                          AdobeDesignLibraryImageElementType,
+                                          AdobeDesignLibraryLayerStyleElementType];
+    syncOnCommit = YES;
+    libraryQueue = [NSOperationQueue mainQueue];
+    autoSyncDownloadedAssets = NO;
+    
+    AdobeLibraryManager *libMgr = [AdobeLibraryManager sharedInstance];
+    libMgr.syncAllowedByNetworkStatusMask = AdobeNetworkReachableViaWiFi | AdobeNetworkReachableViaWWAN;
+    
+    NSString *rootLibDir = NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES)[0];
+    rootLibDir = [rootLibDir stringByAppendingPathComponent:[[NSBundle mainBundle] bundleIdentifier]];
+    rootLibDir = [rootLibDir stringByAppendingPathComponent:@"design-libraries"];
```

Comment:
Small nitpick: This should probably be just “libraries” since the “design” part is the old name.